### PR TITLE
rustfmt: add page (#2318)

### LINKED ISSUE #2318: page request: rustfmt
Add documentation for `rustfmt`

### PR DESCRIPTION
**Pull request by mfrw**
Closes #2318 
----

- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited.

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines.

### CODE REVIEW

File: pages/common/rustfmt.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+- Format a file, overwriting the original file in-place:
+
+`rustfmt {{source.rs}}`
+
+- Check a file for formating and display any changes on console:
+
+`rustfmt --check {{source.rs}}`
+
+- Backup any modified files before formatting:
```
Comment (agnivade) [inline pages/common/rustfmt.md]: What does it backup the file to ? sth like source.rs.bak ?

Comment (mfrw) [reply pages/common/rustfmt.md]: The backup file for `source.rs` -> `source.bk` by default.

Comment (agnivade) [reply pages/common/rustfmt.md]: Let's mention that in the description.

Comment (mfrw) [reply pages/common/rustfmt.md]: sure @agnivade

File: pages/common/rustfmt.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+- Format a file, overwriting the original file in-place:
+
+`rustfmt {{source.rs}}`
+
+- Check a file for formating and display any changes on console:
+
+`rustfmt --check {{source.rs}}`
+
+- Backup any modified files before formatting. If a file source.rs is formated, the backup is source.bk:
```
Comment (sbrl) [inline pages/common/rustfmt.md]: Typo: `formated` -> `formatted`.

Perhaps `source.rs` and `source.bk` should be enclosed in backticks?

Also, the full-stop here is a bit weird. perhaps we could put the second sentence in brackets?

File: pages/common/rustfmt.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# rustfmt
+
+> Tool for formatting Rust source code.
+
+- Format a file, overwriting the original file in-place:
+
+`rustfmt {{source.rs}}`
+
+- Check a file for formating and display any changes on console:
```
Comment (sbrl) [inline pages/common/rustfmt.md]: Typo: `formating` -> `formatting`. Also, perhaps `on console` would read better as `on the console`?

File: pages/common/rustfmt.md (hunk 2 above)
Comment (mfrw) [reply pages/common/rustfmt.md]: sure

File: pages/common/rustfmt.md (hunk 3 above)
Comment (mfrw) [reply pages/common/rustfmt.md]: Ahh ... I agree

File: pages/common/rustfmt.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+- Format a file, overwriting the original file in-place:
+
+`rustfmt {{source.rs}}`
+
+- Check a file for formatting and display any changes on the console:
+
+`rustfmt --check {{source.rs}}`
+
+- Backup any modified files before formatting (if a file `source.rs` is formated, the backup is `source.bk`):
```
Comment (agnivade) [inline pages/common/rustfmt.md]: How about we simplify it to - `(the original file is renamed with a '.bk' extension)`

**Pull request merged**